scrooge-generator/src/main/resources/scalagen/finagleService.scala

```diff
@@ -21,7 +21,8 @@ class {{ServiceName}}$FinagleService(
   iface: {{ServiceName}}[Future],
   protocolFactory: TProtocolFactory,
   stats: StatsReceiver,
-  maxThriftBufferSize: Int
+  maxThriftBufferSize: Int,
+  serviceName: String = "{{ServiceName}}"
```

Comment:
hmm, instead of using a default argument, could you overload the constructor?  I think that will ease binary compatibility.